src/machines/java/com/enderio/machines/common/blockentity/WiredChargerBlockEntity.java

```diff
@@ -0,0 +1,104 @@
+    public static final QuadraticScalable USAGE = new QuadraticScalable(CapacitorModifier.ENERGY_USE, () -> 100f);
+
+    public static final SingleSlotAccess ITEM_TO_CHARGE = new SingleSlotAccess();
+
+    public static final SingleSlotAccess ITEM_CHARGED = new SingleSlotAccess();
+
+    private float progress = 0;
+
+    public WiredChargerBlockEntity(BlockEntityType<?> type, BlockPos worldPosition, BlockState blockState) {
+        super(EnergyIOMode.Input, CAPACITY, TRANSFER, USAGE, type, worldPosition, blockState);
+        addDataSlot(new FloatDataSlot(this::getProgress, p -> progress = p, SyncMode.GUI));
+    }
+
+
+    @Override
+    public MachineInventoryLayout getInventoryLayout() {
+        return MachineInventoryLayout
+            .builder()
+            .capacitor()
+            .inputSlot((slot, stack) -> acceptItem(stack))
+            .slotAccess(ITEM_TO_CHARGE)
+            .outputSlot()
+            .slotAccess(ITEM_CHARGED)
+            .build();
+    }
+
+
+    @Nullable
+    @Override
+    public AbstractContainerMenu createMenu(int pContainerId, Inventory pPlayerInventory, Player pPlayer) {
+        return new WiredChargerMenu(this, pPlayerInventory, pContainerId);
+    }
+
+    @Override
+    public void serverTick() {
+        super.serverTick();
+        chargeItem();
+    }
+
+
+    public boolean acceptItem(ItemStack item) {
+        Optional<IEnergyStorage> energyHandlerCap = item.getCapability(ForgeCapabilities.ENERGY).resolve();
+        return energyHandlerCap.isPresent();
+    }
+
+    public void chargeItem() {
+        ItemStack inputItem = ITEM_TO_CHARGE.getItemStack(this);
+        ItemStack outputItem = ITEM_CHARGED.getItemStack(this);
+        if (!inputItem.isEmpty() && outputItem.isEmpty() && isCapacitorInstalled()) {
+            Optional<IEnergyStorage> energyHandlerCap = inputItem.getCapability(ForgeCapabilities.ENERGY).resolve();
+            if (energyHandlerCap.isPresent()) {
+                IEnergyStorage itemEnergyStorage = energyHandlerCap.get();
+                if (itemEnergyStorage.getEnergyStored() == itemEnergyStorage.getMaxEnergyStored()) {
+                    ITEM_CHARGED.setStackInSlot(this, inputItem);
+                    ITEM_TO_CHARGE.setStackInSlot(this, ItemStack.EMPTY);
+                } else {
+                    // The energyExtracted per tick should increase if the charged item has more energy and with the tiers of the capacitor installed
+                    int energyExtracted = itemEnergyStorage.getMaxEnergyStored() / (1000-this.energyStorage.getMaxEnergyUse());
+                    if (canAct() && this.energyStorage.getEnergyStored() >= energyExtracted) {
+                        itemEnergyStorage.receiveEnergy(energyExtracted, false);
+                        this.energyStorage.takeEnergy(energyExtracted);
+                        this.progress = (float)itemEnergyStorage.getEnergyStored()/itemEnergyStorage.getMaxEnergyStored();
```

Comment:
Okay so I might have suggested moving the charging logic into a "PoweredTask" but because I'm reworking this system I'm happy to let this in for now. But I'll explain so you're aware for future :)

To make machine logic slightly clearer, we have a "PoweredTask" class which represents a single unit of work. Then the machine will see if a new task can be started, and if it can then it will. This task will simply (in the case of the charger) take energy from the machine and add it to the item every tick, and report progress (this progress is also automatically synced over the network). This allows units of work to be easy to define and also makes the block entity logic a bit easier to read imo.

Reply:
So in essence, this isn't something to fix in this specific PR because of the rework I'm doing, but its good to know for future :)

That being said, its possible the "task" approach isn't applicable to this, so happy for you or @agnor99 or @ferriarnus to tune in as well :)

Reply:
I will let this for now. Feel free to use your new task system for more clarity.